File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/finder/Finder.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.finder;

import ru.ioleksiv.telegram.bot.core.controller.handler.check.Validator;
import ru.ioleksiv.telegram.bot.core.model.ITelegram;

import java.lang.annotation.Annotation;

public interface Finder {
    <SA extends Annotation, T> Validator<T> find(SA annotation);

    <T extends ITelegram> Validator<T> find(String validatorName, Class<T> targetClass);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/core/controller/annotations/parser/finder/Finder.kt
package io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.finder

import io.github.oleksivio.tl.kbot.core.core.controller.handler.check.Validator
import io.github.oleksivio.tl.kbot.server.api.model.ITelegram
import kotlin.reflect.KClass

interface Finder {
    fun <SA : Annotation, T> find(annotation: SA): Validator<T>

    fun <T : ITelegram> find(validatorName: String, targetClass: KClass<T>): Validator<T>

}
